var/spack/repos/builtin/packages/steps/package.py

```diff
@@ -42,13 +46,15 @@ class Steps(CMakePackage):
     depends_on("py-scipy", type=("build", "test"))
     depends_on("py-matplotlib", type=("build", "test"))
     depends_on("python")
-
+    depends_on("py-cmake-format", type="build", when="+codechecks")
+    depends_on('py-pre-commit', type='build', when='+codechecks')
+    depends_on('py-pyyaml', type='build', when='+codechecks')
```

Comment:
Stupid question. Shouldn't you enable something explicitly when using `+codechecks`? Maybe I just need more :coffee: …

Reply:
I do not understand the question, sorry. Maybe I just need more ☕

Reply:
You added a variant and dependencies, but no additional cmake flags?

Reply:
I.e., in the code below, I'd expect a

```
if "+codechecks" in spec:
```

Reply:
> I.e., in the code below, I'd expect a
> 
> ```
> if "+codechecks" in spec:
> ```

Good point, but that will come later if that's ok for you.